client/utils/hash.js

```diff
@@ -0,0 +1,31 @@
+const memory = {};
+
+export default function hash(obj) {
```

Comment:
1. Is this used anywhere? I don't see an import for it.
2. We have `sha.js` as a dependency already, so we should use that for hashing instead of writing our own.